pkg/operator/sync_test.go

```diff
@@ -154,3 +158,27 @@ func withCABundle(caBundle string) kubeCloudConfigOption {
 		kubeCloudConfig.Data["ca-bundle.pem"] = caBundle
 	}
 }
+
+func TestSyncCustomResourceDefinitions(t *testing.T) {
+	kubeClient := fake.NewSimpleClientset()
+	//	sharedInformer := informers.NewSharedInformerFactory(client, 0)
+	//	cmInformer := sharedInformer.Core().V1().ConfigMaps()
+	optr := &Operator{
+		//		clusterCmLister: cmInformer.Lister(),
+		libgoRecorder: events.NewRecorder(kubeClient.CoreV1().Events("openshift-machine-config-operator"), "machine-config-operator", &corev1.ObjectReference{}),
+		apiExtClient: ???,
+	}
+	optr.syncCustomResourceDefinitions()
+	crds := []string{
+		"manifests/controllerconfig.crd.yaml",
+	}
+
+	for _, crd := range crds {
+		crdBytes, err := manifests.ReadFile(crd)
+		assert.Nil(t, err, fmt.Errorf("error getting asset %s: %v", crd, err))
+		expectedCrd := resourceread.ReadCustomResourceDefinitionV1Beta1OrDie(crdBytes)
+		actualCrd, err := optr.crdLister.Get(expectedCrd.Name)
+		assert.Nil(t, err)
+		assert.Equal(t, expectedCrd, actualCrd)
```

Comment:
@yuqi-zhang when you look at what we are ensuring with the old functions, do I need to make more specific assertions about what `syncCustomResourceDefinitions()` should be doing here?

Reply:
Answered in chat, but to also summarize here, we should aim to test with mock definitions of what fields we care about in the various MCO resource definitions, to ensure creation/updates/deletion is handled correctly